src/windows/Link.tsx

```diff
@@ -0,0 +1,149 @@
+
+    private _onFocusableRef = (btn: RNW.FocusableWindows | null): void => {
+        this._focusableElement = btn;
+    }
+
+    render() {
+
+        // Fallback to native-common fast if the keyboard enabled component is not available
+        if (!HasFocusableWindows) {
+            return super.render();
+        }
+
+        let content: JSX.Element = super.render();
+
+        // The "in text parent" case requires a special nyi control.
+        if (this.context && !this.context.isRxParentAText) {
+
+            let tabIndex: number | undefined = this.getTabIndex();
+            let windowsTabFocusable: boolean =  tabIndex !== undefined && tabIndex >= 0;
+
+            // RNW.FocusableWindows doesn't participate in layouting, it basically mimics the position/size of the child
+            let focusableViewProps: RNW.FocusableProps = {
+                ref: this._onFocusableRef,
+                isTabStop: windowsTabFocusable,
+                tabIndex: tabIndex,
+                disableSystemFocusVisuals: false,
+                handledKeyDownKeys: DOWN_KEYCODES,
+                handledKeyUpKeys: UP_KEYCODES,
+                onKeyDown: this._onKeyDown,
+                onKeyUp: this._onKeyUp,
+                onFocus: this._onFocus,
+            };
+
+            content = (
+                <RNW.FocusableWindows
+                    {...focusableViewProps}
+                >
+                    {content}
+                </RNW.FocusableWindows>
+            );
+        }
+
+        return content;
+    }
+
+    focus() {
+        if (this._focusableElement &&
+            this._focusableElement.focus) {
+                this._focusableElement.focus();
+        }
+    }
+
+    blur() {
+        if (this._focusableElement &&
+            this._focusableElement.blur) {
+                this._focusableElement.blur();
+        }
+    }
+
+    private _onKeyDown = (e: React.SyntheticEvent<any>): void => {
+        if (this.props.onPress) {
+            let keyEvent = EventHelpers.toKeyboardEvent(e);
+            let key = keyEvent.keyCode;
+            // ENTER triggers press on key down
+            if (key === KEY_CODE_ENTER) {
+                // Defer to base class
+                this._onPress(keyEvent);
+            }
+        }
+    }
+
+    private _onKeyUp = (e: React.SyntheticEvent<any>): void => {
+        if (this.props.onPress) {
+            let keyEvent = EventHelpers.toKeyboardEvent(e);
+            if (keyEvent.keyCode === KEY_CODE_SPACE) {
+                 // Defer to base class
+                this._onPress(keyEvent);
+            }
+        }
+    }
+
+    private _onFocus = (e: React.SyntheticEvent<any>): void => {
+        this.onFocus();
+    }
+
+    // From FocusManagerFocusableComponent interface
+    //
+    onFocus() {
+        // Focus Manager hook
+    }
+
+    getTabIndex(): number | undefined {
+        // Focus Manager may override this
+        return 0;
```

Comment:
Why don't we reference `this.props.tabIndex`? It looks like it's a part of `Types.LinkProps`.

Reply:
Aha, The 0.42 RXP didn't have it. 
The current one does. Weirdly enough this tabIndex was added to LinkProps instead of making the LinkProps derive from that CommonAccessibilityProps or so..

---
In reply to: [159584978](https://github.com/Microsoft/reactxp/pull/404#discussion_r159584978) [](ancestors = 159584978)